src/requestCertificate.jsx

```diff
@@ -160,6 +161,108 @@ const KeyFileRow = ({ setKeyFile, mode }) => {
     );
 };
 
+export const ResubmitCertificateModal = ({ onClose, cas, addAlert, cert, certPath, mode }) => {
+    const [ca, setCa] = useState(getCAName(cas, cert));
+    const [subjectName, setSubjectName] = useState(cert.subject && cert.subject.v);
+    const [dnsName, _setDnsName] = useState(cert.hostname && cert.hostname.v.join(','));
+    const [principalName, setPricipalName] = useState(cert.principal && cert.principal.v.join(','));
+    const [errorName, setErrorName] = useState("");
+    const [errorMessage, setErrorMessage] = useState("");
+
+    const setDnsName = (value) => {
+        let newValue = value.replace(/\s/g, ',');
+        while (newValue.includes(",,"))
+            newValue = value.replace(",,", ',');
+
+        _setDnsName(newValue);
+    };
+
+    const onResubmit = () => {
+        const casKeys = Object.keys(cas);
+        let caPath;
+        casKeys.forEach(key => {
+            if (cas[key].nickname.v === ca)
+                caPath = key;
+        });
+
+        let subjectNameParam = subjectName;
+        if (subjectNameParam && !subjectNameParam.includes("="))
+            subjectNameParam = "CN=" + subjectNameParam;
+        let dnsNameParam;
+        if (dnsName) {
+            dnsNameParam = dnsName.split(',');
+            dnsNameParam = dnsNameParam.filter(Boolean); // Removes empty string entries
+        }
+
+        const parameter = { ca: cockpit.variant("s", caPath) };
+        parameter["template-subject"] = subjectNameParam
+            ? cockpit.variant("s", subjectNameParam)
+            : cockpit.variant("s", "");
```

Comment:
This can be written in an easier way like this:

    parameter["template-subject"] = cockpit.variant("s", subjectNameParam || "")

same for the two others below.